src/include/reconcile_inline.h

```diff
@@ -371,16 +365,14 @@ __wt_rec_cell_build_addr(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_ADDR *add
         WT_ASSERT(session, addr != NULL);
         val->buf.data = addr->addr;
         val->buf.size = addr->size;
-        val->cell_len =
-          __wt_cell_pack_addr(session, &val->cell, cell_type, recno, &addr->ta, val->buf.size);
     } else {
         WT_ASSERT(session, addr == NULL);
         val->buf.data = vpack->data;
         val->buf.size = vpack->size;
-        val->cell_len =
-          __wt_cell_pack_addr(session, &val->cell, cell_type, recno, &vpack->ta, val->buf.size);
     }
 
+    val->cell_len =
+      __wt_cell_pack_addr(session, &val->cell, cell_type, recno, page_del, ta, val->buf.size);
```

Comment:
Shall we add an assert here to ensure that cell_type is `WT_CELL_ADDR_DEL` when the `page_del` is passed? The main reason behind this is as we access the page_del members if the cell type is DEL in unpack and also we set every internal page with WT_PAGE_FT_UPDATE

Reply:
In that function, `__wt_rec_cell_build_addr()`, the passed in cell_type will be either `WT_CELL_ADDR_LEAF` or `WT_CELL_ADDR_LEAF_NO`, we switch to the `WT_CELL_ADDR_DEL` cell type inside `__wt_cell_pack_addr()`.

The `__wt_cell_pack_addr()` function does assert the cell type is LEAF or LEAF_NO whenever `page_del` isn't NULL.

Please let me know if I'm misunderstanding your comment.

Reply:
That reminds me (forgot to mention this yesterday): that assert should be restricted to `WT_CELL_ADDR_LEAF_NO`; we can't (and don't) fast-delete pages with overflow items and the `NO` is what's used to check that further up the call chain.

Reply:
Thanks, fixed in eb90fd8.